gbm k-fold CV while selecting best number of trees

dmcartor opened the pull request:
added a function to conduct gbm k-fold cross-validation over a set of metaparameters in parallel while also selecting the optimal number of trees in a data-driven manner

Commit:
added a function to conduct gbm k-fold cross-validation over a set of metaparameters in parallel while also selecting the optimal number of trees in a data-driven manner

patr1ckm commented:
Awesome! This code looks great, as do the tests. 

Do you mind making it pass R CMD Check (no notes)? See issues [here](https://travis-ci.org/patr1ckm/mvtboost/builds/203547153). Looks like it's just  adding `@importFrom stats runif`
`@importFrom gbm gbm.fit, gbm_more` 
 removing `require(gbm)` and `require(parallel)`, and making some easy changes to the example docs. 

 I think if you commit and `git push origin master` on your fork, this pull request will update.

Commit:
minor changes to address R CMD CHECK NOTEs; added a hidden print.summary.gbmcverr function

dmcartor commented:
These issues have been addressed, and I've updated the pull request. I didn't get any ERRORs, WARNINGs, or NOTEs on my personal results from R CMD CHECK that relate to what I've added, but there was one ERROR based on the tests of metb.fit() that we discussed earlier.

patr1ckm commented:
Looks great!
